scrapy/core/downloader/contextfactory.py

```diff
@@ -17,4 +23,6 @@ def getContext(self, hostname=None, port=None):
         # Enable all workarounds to SSL bugs as documented by
         # http://www.openssl.org/docs/ssl/SSL_CTX_set_options.html
         ctx.set_options(SSL.OP_ALL)
+        if hostname and ClientTLSOptions is not None: # workaround for TLS SNI
+            ClientTLSOptions(hostname, ctx)
```

Comment:
I'd merge this and put in the wild under unstable branch so we can get more feedback.